packages/frontend-app-api/src/wiring/createApp.test.tsx

```diff
@@ -0,0 +1,240 @@
+
+describe('createInstances', () => {
+  it('throws an error when a root extension is parametrized', () => {
+    const config = new MockConfigApi({
+      app: {
+        extensions: [
+          {
+            root: {
+              at: '',
+            },
+          },
+        ],
+      },
+    });
+    const plugins = [
+      createPlugin({
+        id: 'plugin',
+        extensions: [],
+      }),
+    ];
+    expect(() => createInstances({ config, plugins })).toThrow(
+      'A "root" extension was detected on the config file and root extensions are not configurable',
+    );
+  });
+
+  it('throws an error when a root extension is overridden', () => {
+    const config = new MockConfigApi({});
+    const plugins = [
+      createPlugin({
+        id: 'plugin',
+        extensions: [
+          createExtension({
+            id: 'root',
+            at: 'core.routes/route',
+            inputs: {},
+            output: {},
+            factory() {},
+          }),
+        ],
+      }),
+    ];
+    expect(() => createInstances({ config, plugins })).toThrow(
+      'The following plugins are overriding root extensions and root extensions cannot be overridden: plugin',
+    );
+  });
+
+  describe('throws an error when immediate cyclical dependencies are found', () => {
+    it('in an immediate level (e.g., A(plugin.page) -> A(plugin.page))', () => {
+      const config = new MockConfigApi({});
+
+      const addonExtensionData =
+        createExtensionDataRef<JSX.Element>('plugin.page.addon');
+
+      const addon = createExtension({
+        id: 'plugin.page', // cyclical
+        at: 'plugin.page/addons',
+        inputs: {},
+        output: {
+          element: addonExtensionData,
+        },
+        factory({ bind }) {
+          bind({
+            element: <div key="addon">Addon</div>,
+          });
+        },
+      });
+
+      const page = createPageExtension({
+        id: 'plugin.page',
+        defaultPath: '/',
+        routeRef: createRouteRef({ id: 'plugins.page.addon' }),
+        inputs: {
+          addons: createExtensionInput({
+            element: addonExtensionData,
+          }),
+        },
+        loader: async ({ inputs }) => (
+          <div>Page {inputs.addons.map(({ element }) => element)}</div>
+        ),
+      });
+
+      const plugins = [
+        createPlugin({
+          id: 'plugin',
+          extensions: [page, addon],
+        }),
+      ];
+
+      expect(() => createInstances({ config, plugins })).toThrow(
+        'There is a cyclical dependency with the extension "plugin.page": core.layout → core.routes → plugin.page → plugin.page',
+      );
+    });
+
+    it('in an intermediate level (e.g., A(core.routes) -> B(plugin.page) -> A(core.routes))', () => {
+      const config = new MockConfigApi({});
+
+      const addonExtensionData =
+        createExtensionDataRef<JSX.Element>('plugin.page.addon');
+
+      const addon = createExtension({
+        id: 'core.routes', // cyclical
```

Comment:
For clarity I'm thinking it's best that these tests don't rely on the existing core extensions, but instead create their own simple extension structure with very boring input/output and names. I had a bit of a hard time figuring out how this test even manages to cause a cycle with the mix of core overrides and createPageExtension 😅

Reply:
Yes, I tried to avoid creating another extension and you got it 😬.
I'll update the tests 🙌🏻

Reply:
Done 🙂 : https://github.com/backstage/backstage/pull/20032/commits/054d7aceb2e7e4b444a343f39d78ec6877a6df39